src/components/ScholarshipList.jsx

```diff
@@ -26,32 +24,6 @@ const useStyles = makeStyles(() => ({
 
 function ScholarshipList({ noResultsNode, listFn }) {
   const classes = useStyles();
-  const [shareDialogOpen, setShareDialogOpen] = useState(false);
-  const closeShareDialog = () => setShareDialogOpen(false);
-
-  const [shareSiteLink, setShareSiteLink] = React.useState('');
-  const [shareSiteTitle, setShareSiteTitle] = React.useState('');
-
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
-  const shareFn = (id, data) => () => {
-    const title = `${data.amount} - ${data.name} | ${BRAND_NAME}`;
-    const url = `https://${window.location.hostname}/scholarships/${id}`;
-    const text =
-      `${data.amount} - ${data.name} | ${BRAND_NAME}\n` +
-      `${data.deadline.toLocaleDateString()}\n`;
-    if (navigator.share) {
-      navigator
-        .share({ title, url, text })
-        // eslint-disable-next-line no-console
-        .then(() => console.log('Thanks for sharing!'))
-        // eslint-disable-next-line no-console
-        .catch(console.error);
-    } else {
-      setShareSiteLink(url);
-      setShareSiteTitle(title);
-      setShareDialogOpen(true);
-    }
-  };
```

Comment:
You could also keep this in the DetailCard and in the `else` case do the `history.replace` you ended up doing. I think where possible it may be better to use the native share dialog as it usually redirects to another app anyway.

Reply:
Wouldn't a better place for this be in `ShareDialog`

Reply:
Hmm, potentially but in that case in `ShareDialog` you'll want to render nothing (`</>`) if the navigator supports sharing natively.

Reply:
For now I just left it in `ScholarshipDetailCard`